rotkehlchen/data_migrations/migrations/migration_12.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.data_import.importers.constants import ROTKI_EVENT_PREFIX
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.data_migrations.progress import MigrationProgressHandler
+    from rotkehlchen.rotkehlchen import Rotkehlchen
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+PREFIX = f'{ROTKI_EVENT_PREFIX}_%'
+CHANGES = [  # TO TYPE, TO SUBTYPE, FROM TYPE, FROM SUBTYPE
+    (HistoryEventType.DEPOSIT, HistoryEventSubType.NONE, HistoryEventType.DEPOSIT, HistoryEventSubType.SPEND),  # noqa: E501
+    (HistoryEventType.WITHDRAWAL, HistoryEventSubType.NONE, HistoryEventType.WITHDRAWAL, HistoryEventSubType.RECEIVE),  # noqa: E501
+    (HistoryEventType.RECEIVE, HistoryEventSubType.NONE, HistoryEventType.RECEIVE, HistoryEventSubType.RECEIVE),  # noqa: E501
+
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, HistoryEventType.DEPOSIT, HistoryEventSubType.FEE),  # noqa: E501
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, HistoryEventType.WITHDRAWAL, HistoryEventSubType.FEE),  # noqa: E501
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, HistoryEventType.RECEIVE, HistoryEventSubType.FEE),  # noqa: E501
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, HistoryEventType.STAKING, HistoryEventSubType.FEE),  # noqa: E501
+]
+
+
+def data_migration_12(rotki: 'Rotkehlchen', progress_handler: 'MigrationProgressHandler') -> None:
+    """
+    Migrate rotki events that were broken due to https://github.com/rotki/rotki/issues/6550
+
+    May be introduced at v1.30.2. In any case will be removed when DB upgrade 39 -> 40 happens
+    as it will move to that DB upgrade.
+    """
+    log.debug('Enter data_migration_12')
+    progress_handler.set_total_steps(1)
+    progress_handler.new_step('Migrating custom rotki events')
+
+    with rotki.data.db.conn.write_ctx() as write_cursor:
+        for to_type, to_subtype, from_type, from_subtype in CHANGES:
+            write_cursor.execute(
+                'UPDATE history_events SET type=?, subtype=? WHERE '
+                'event_identifier LIKE ? AND type=? AND subtype=?',
+                (
+                    to_type.serialize(), to_subtype.serialize(), PREFIX,
+                    from_type.serialize(), from_subtype.serialize(),
+                ),
+            )
```

Comment:
why not make this a executemany?

Reply:
Does it work? Let me check. For some reason I think it did not.

Reply:
you need to create the list of serialized values but otherwise it should be faster than len(changes) executes